### PR TITLE
WMTS: Map CURLE_FILE_COULDNT_READ_FILE  to 404 (fixes #2941)

### LINKED ISSUE #2941: WMTS: local files do not accept ZeroBlockHttpCodes
## Expected behavior and actual behavior.

I'm consuming a local WMTS server structure, for some technical reasons some tiles may be missing.

Local WMTS with missing tiles will generate and error and a blank image.
I expected it to  behave like online WMTS, an image with missing areas.

## Steps to reproduce the problem.

Draw a local WMTS with missing tiles.

## Operating system

The problem is system independent

## GDAL version and provenance

Build from source 2.4.0

## Proposed solution

From the documentation, I understand the `ZeroBlockHttpCodes` is used in this cases, eg.: 404 - file not found in a online server.

But for local files, `curl` does not set a Http Response code, currently gdal already goes around it and when no error message is set, [it sets the response code to 200](https://github.com/OSGeo/gdal/blob/master/gdal/frmts/wms/gdalhttp.cpp#L224).

A solution would be map the curl error `CURLE_FILE_COULDNT_READ_FILE` to a gdal reponse code `404`.

*In case we agree upon the solution I plan to make a pull request to the current version.*

### PR DESCRIPTION
**Pull request by janosimas**
## What does this PR do?
Identify local wmts tiles not found with CURLE_FILE_COULDNT_READ_FILE and map to http status 404

## What are related issues/pull requests?

fixes #2941

## Tasklist

 - [x] Map CURLE_FILE_COULDNT_READ_FILE  to 404
 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

 Built and tested on windows.

### CODE REVIEW

File: gdal/frmts/wms/gdalhttp.cpp [hunk 1]
```diff
@@ -67,6 +67,29 @@ static size_t WriteFunc(void *buffer, size_t count, size_t nmemb, void *req) {
     return nmemb;
 }
 
+// Process curl errors
+static void ProcessCurlErros(CURLMsg* msg, WMSHTTPRequest* pasRequest, int nRequestCount)
```
Comment (rouault) [inline gdal/frmts/wms/gdalhttp.cpp:71]:
```suggestion
static void ProcessCurlErrors(CURLMsg* msg, WMSHTTPRequest* pasRequest, int nRequestCount)
```

File: gdal/frmts/wms/gdalhttp.cpp [hunk 2]
```diff
@@ -177,6 +200,8 @@ CPLErr WMSHTTPFetchMulti(WMSHTTPRequest *pasRequest, int nRequestCount) {
         do {
             CURLMsg *m = curl_multi_info_read(curl_multi, &msgs_in_queue);
             if (m && (m->msg == CURLMSG_DONE)) {
+                ProcessCurlErros(m, pasRequest, nRequestCount);
```
Comment (rouault) [inline gdal/frmts/wms/gdalhttp.cpp:203]:
```suggestion
                ProcessCurlErrors(m, pasRequest, nRequestCount);
```

File: gdal/frmts/wms/gdalhttp.cpp [hunk 3]
```diff
@@ -194,6 +219,19 @@ CPLErr WMSHTTPFetchMulti(WMSHTTPRequest *pasRequest, int nRequestCount) {
...
     } while (still_running || conn_i != nRequestCount);
+
+    // process any message still in queue
+    CURLMsg* msg;
+    int msgs_in_queue;
+    do {
+        msg = curl_multi_info_read(curl_multi, &msgs_in_queue);
+        if (msg != nullptr) {
+            if (msg->msg == CURLMSG_DONE) {
+                ProcessCurlErros(msg, pasRequest, nRequestCount);
```
Comment (rouault) [inline gdal/frmts/wms/gdalhttp.cpp:230]:
```suggestion
                ProcessCurlErrors(msg, pasRequest, nRequestCount);
```

**Comment by rouault** [comment]
Would you mind adding a test case for this in autotest/gdrivers/wms.py ?

**Comment by janosimas** [comment]
I'll take a look at it.

**Comment by rouault** [comment]
@janosimas ping

**Comment by janosimas** [comment]
Sorry @rouault, I wont be able to look at it until next month or the other.
I got a delivery that is taking all my time.

**Comment by janosimas** [comment]
I wasn't able to setup a test environment, I'll need some help there.

I was able to build gdal:
```bash
nmake /f makefile.vc
nmake /f makefile.vc install
```

and generate the python bindings:
```bash
nmake /f makefile.vc bindings
nmake /f makefile.vc bindings_install
```

I also installed the generated gdal `egg`.

After that I'm kind of lost, when I import (or `pytest`) it finds the `osgeo` package but fails the importing. I suppose it's not finding the `dll`s.
```
>>> import osgeo
Traceback (most recent call last):
  File "C:\Users\simjan\AppData\Local\Programs\Python\Python39-32\lib\site-packages\gdal-3.1.0-py3.9-win32.egg\osgeo\__init__.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Users\simjan\AppData\Local\Programs\Python\Python39-32\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 565, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1108, in create_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
ImportError: DLL load failed while importing _gdal: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simjan\AppData\Local\Programs\Python\Python39-32\lib\site-packages\gdal-3.1.0-py3.9-win32.egg\osgeo\__init__.py", line 32, in <module>
    _gdal = swig_import_helper()
  File "C:\Users\simjan\AppData\Local\Programs\Python\Python39-32\lib\site-packages\gdal-3.1.0-py3.9-win32.egg\osgeo\__init__.py", line 30, in swig_import_helper
    raise ImportError(traceback_string + '\n' + msg)
ImportError: Traceback (most recent call last):
  File "C:\Users\simjan\AppData\Local\Programs\Python\Python39-32\lib\site-packages\gdal-3.1.0-py3.9-win32.egg\osgeo\__init__.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Users\simjan\AppData\Local\Programs\Python\Python39-32\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 565, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1108, in create_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
ImportError: DLL load failed while importing _gdal: The specified module could not be found.

On Windows, with Python >= 3.8, DLLs are no longer imported from the PATH.
If gdalXXX.dll is in the PATH, then set the USE_PATH_FOR_GDAL_PYTHON=YES environment variable
to feed the PATH into os.add_dll_directory().
```

I tried to add `USE_PATH_FOR_GDAL_PYTHON=YES` to the environment variables but I got other errors related to windows paths.

Any tips there?

**Pull request closed without merging**